### PR TITLE
(TC-154) Capture Docker API Version in Reports

### PR DESCRIPTION
**Pull request by jfryman**
This PR adds a new capability, Docker Server, which polls the host for Version information associated with Docker for publishing as part of a report run.

Supersedes #50 

screenshot_2017-06-19_15-41-21

### CODE REVIEW

File: capabilities/docker/docker.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	v := client.ServerVersion(ctx)
+
+	return map[string]interface{}{
+		"APIVersion":    v.APIVersion,
+		"MinAPIVersion": v.MinAPIVersion,
+	}
+}
+
+func init() {
+	logging.Debug("[Docker Server] Initialising capability: %s", dockerCapability.Title)
```
Comment (jfryman) [inline capabilities/docker/docker.go]: Note: For the time being, I went with HRH's English. Just to keep consistency until we go one way or the other.

File: capabilities/docker/docker.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		logging.Debug("[Docker Server] Harvesting docker server information associated with %s [%s]", target.Name, target.ID)
+		capability.HarvestID = id
+
+		filtered, _ := payloadfilter.Filter(InfoToMap(client))
+
+		capability.Payload = filtered
+	},
+}
+
+// InfoToMap takes the Docker Context and returns a formatted map[string]interface{} contaning
```
Comment (johnmccabe) [inline capabilities/docker/docker.go]: spelling

File: capabilities/docker/docker.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+// information exposted via the Docker API
+func InfoToMap(client dockeradapter.Harvester) map[string]interface{} {
+	ctx := context.Background()
+	v := client.ServerVersion(ctx)
+
+	return map[string]interface{}{
+		"APIVersion":    v.APIVersion,
+		"MinAPIVersion": v.MinAPIVersion,
+	}
```
Comment (johnmccabe) [inline capabilities/docker/docker.go]: This doesn't actually expose the Server version? Just the Min/Max API versions, is this a deliberate choice?

File: capabilities/docker/docker.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package docker
```
Comment (johnmccabe) [inline capabilities/docker/docker.go]: I'd be inclined to use a less generic package name/id for this capability, not a blocker tho.

File: capabilities/docker/docker.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+// InfoToMap takes the Docker Context and returns a formatted map[string]interface{} contaning
+// information exposted via the Docker API
+func InfoToMap(client dockeradapter.Harvester) map[string]interface{} {
+	ctx := context.Background()
+	v := client.ServerVersion(ctx)
+
+	return map[string]interface{}{
+		"APIVersion":    v.APIVersion,
```
Comment (johnmccabe) [inline capabilities/docker/docker.go]: This could blow up if c.Client.ServerVersion(ctx) returns an error. Best to bubble the error up and handle it here.

**Comment by johnmccabe** [comment]
Thanks @jfryman just a few comments.

File: capabilities/docker/docker.go (hunk 1 above)
Comment (johnmccabe) [inline capabilities/docker/docker.go]: tumblr_mcmpyduoqd1r3i95ho6_r1_500

File: capabilities/docker/docker.go (hunk 4 above)
Comment (jfryman) [inline capabilities/docker/docker.go]: I did get stuck here... names gladly acceped. I originally though "host", but that's already in use and does not adequately allow for future cases like K8s or swarm. I did consider `dockerhost` or `dockerserver`, but both also felt overloaded.

Naming things is hard!

**Comment by jfryman** [comment]
Think I got all those comments, @johnmccabe... lmk if there is more or merge if :+1:

File: capabilities/docker/docker.go (hunk 4 above)
Comment (johnmccabe) [inline capabilities/docker/docker.go]: Happy to leave this as is, can refactor when inspiration strikes for someone

**Pull request merged**